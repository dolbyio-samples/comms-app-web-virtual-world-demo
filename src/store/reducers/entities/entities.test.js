import { describe, expect, it } from "vitest";

import reducer from "./entities";

describe("entities test suite", () => {
  it("dispatch leave conference action, reducer should return initial state", () => {
    // 1- arrange
    const action = { type: "LEAVE_CONFERENCE" };
    const state = {};

    // 2- act
    const updatedState = reducer(state, action);

    // 3- assert
    expect(updatedState).toMatchSnapshot();
  });

  it("dispatch add participant action, reducer should add one participant entity", () => {
    // 1- arrange
    const action = {
      type: "ADD_PARTICIPANT",
      payload: {
        entityType: "participants",
        data: { participantId: { name: "toto" } },
      },
    };
    const state = {};

    // 2- act
    const updatedState = reducer(state, action);

    // 3- assert
    expect(updatedState).toMatchSnapshot();
  });

  it("dispatch update participant action, reducer should add one participant entity", () => {
    // 1- arrange
    const action = {
      type: "UPDATE_PARTICIPANT",
      payload: {
        entityType: "participants",
        id: "participantId",
        data: { surname: "titi" },
      },
    };

    const state = {
      participants: {
        participantId: {
          name: "toto",
        },
      },
    };

    // 2- act
    const updatedState = reducer(state, action);

    // 3- assert
    expect(updatedState).toMatchSnapshot();
  });

  it("dispatch remove participant action, reducer should remove one participant entity", () => {
    // 1- arrange
    const action = {
      type: "REMOVE_PARTICIPANT",
      payload: {
        entityType: "participants",
        id: "participantId",
      },
    };

    const state = {
      participants: {
        participantId: {
          name: "toto",
        },
      },
    };

    // 2- act
    const updatedState = reducer(state, action);

    // 3- assert
    expect(updatedState).toMatchSnapshot();
  });
});
