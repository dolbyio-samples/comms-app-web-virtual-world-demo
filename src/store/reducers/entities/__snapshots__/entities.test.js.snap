// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`entities test suite > dispatch add participant action, reducer should add one participant entity 1`] = `
{
  "participants": {
    "participantId": {
      "name": "toto",
    },
  },
}
`;

exports[`entities test suite > dispatch leave conference action, reducer should return initial state 1`] = `{}`;

exports[`entities test suite > dispatch remove participant action, reducer should remove one participant entity 1`] = `
{
  "participants": {},
}
`;

exports[`entities test suite > dispatch update participant action, reducer should add one participant entity 1`] = `
{
  "participants": {
    "participantId": {
      "name": "toto",
      "surname": "titi",
    },
  },
}
`;
